tests/lax_numpy_test.py

```diff
@@ -1978,6 +1978,119 @@ def testMeshGrid(self, shapes, dtype, indexing, sparse, rng):
     lnp_fun = partial(lnp.meshgrid, indexing=indexing, sparse=sparse)
     self._CompileAndCheck(lnp_fun, args_maker, check_dtypes=True)
 
+  def assertDowncastDtypeEqual(self, x, y):
+    """Heuristic for comparing numpy and jax downcast dtypes."""
+    x_dt = jtu._dtype(x)
+    y_dt = jtu._dtype(y)
+    testing_tpu = jtu.device_under_test().startswith("tpu")
+    testing_x32 = not jax.config.read('jax_enable_x64')
+    to32dtype = {onp.int64: onp.int32, onp.uint64: onp.uint32,
+                 onp.float64: onp.float32, onp.float128: onp.float32,
+                 onp.complex128: onp.complex64, onp.complex256: onp.complex64}
+    to32dtype = {onp.dtype(k): onp.dtype(v) for k,v in to32dtype.items()}
+    if testing_tpu or testing_x32:
+      x_dt = to32dtype.get(x_dt, x_dt)
+      y_dt = to32dtype.get(y_dt, y_dt)
+    assert x_dt == y_dt, "truncated dtypes %s != %s" % (x_dt, y_dt)
+
+  @parameterized.named_parameters(
+      jtu.cases_from_list(
+        {"testcase_name": ("_start_shape={}_stop_shape={}_num={}_endpoint={}"
+                           "_retstep={}_dtype={}").format(
+            start_shape, stop_shape, num, endpoint, retstep, dtype),
+         "start_shape": start_shape, "stop_shape": stop_shape,
+         "num": num, "endpoint": endpoint, "retstep": retstep,
+         "dtype": dtype, "rng": rng}
+        for start_shape in [(), (2,), (2, 2)]
+        for stop_shape in [(), (2,), (2, 2)]
+        for num in [5, 20]
+        for endpoint in [True, False]
+        for retstep in [True, False]
+        for dtype in number_dtypes + [None,]
+        for rng in [jtu.rand_default()]))
+  def testLinspace(self, start_shape, stop_shape, num, endpoint,
+                   retstep, dtype, rng):
+    start = rng(start_shape, dtype)
+    stop = rng(stop_shape, dtype)
+    ndim = len(onp.shape(start + stop))
+    # relax default tolerances slightly
+    tol = tolerance(dtype if dtype else onp.float32) * 10
+    for axis in range(0, ndim):
```

Comment:
Test negative values for `axis`? Same for the other tests.

Reply:
thanks for the catch! fixed.